src/protocol/iobuf.c

```diff
@@ -417,17 +413,18 @@ int _z_wbuf_write(_z_wbuf_t *wbf, uint8_t b)
         _z_iosli_write(ios, b);
         return 0;
     }
-    else if (wbf->is_expandable)
+
+    if (wbf->is_expandable)
     {
-        __z_wbuf_new_iosli(wbf, wbf->capacity);
+        __z_wbuf_new_iosli(wbf, ZN_IOSLICE_SIZE);
+        wbf->w_idx++;
```

Comment:
`w_idx` increment should be managed by the functions actually adding/creating the `iosli` to the `wbuf`,

Reply:
done